File: javascript/test/formatHelper.test.js
// This file is part of an MIT-licensed project: see LICENSE file or README.md for details.
// Copyright (c) 2024 Ian Thomas

import {loadTestFile} from '../test/testUtils.js';
import {strict as assert} from 'assert';
import {fountainToHtml} from '../src/formatHelper.js';

describe('FormatHelper', () => {

    describe('Formatted', () => {
        it('should match', () => {
    
          const source = loadTestFile('Formatted.fountain');
          const match = loadTestFile('Formatted.txt');
    
          let formattedText = fountainToHtml(source);
          
          //console.log(formattedText);
          assert.equal(match, formattedText);
        });
    });
    
});

File: csharp/ScreenplayTools.Tests/Fountain/FormatHelperTest.cs
// This file is part of an MIT-licensed project: see LICENSE file or README.md for details.
// Copyright (c) 2024 Ian Thomas

namespace ScreenplayTools.Tests;
using System.IO;
using ScreenplayTools.Fountain;

public class FormatHelperTest
{
    private string loadTestFile(string fileName) {
        return File.ReadAllText("../../../../../tests/"+fileName);
    }

    [Fact]
    public void Formatting()
    {
        string source = loadTestFile("Formatted.fountain");
        string match = loadTestFile("Formatted.txt");

        string formattedText = FormatHelper.FountainToHtml(source);
        //Console.WriteLine(formattedText);

        Assert.Equal(match, formattedText);
    }
}
